src/Platform/Microsoft.Testing.Platform/Services/CurrentTestApplicationModuleInfo.cs

```diff
@@ -95,7 +95,7 @@ public string GetProcessPath()
 #else
     {
         using IProcess currentProcess = process.GetCurrentProcess();
-        return currentProcess.MainModule.FileName;
+        return currentProcess.MainModule?.FileName;
```

Comment:
Let's respect the `throwOnNull` in this code path as well.

```suggestion
        string processPath = currentProcess.MainModule?.FileName;
        ApplicationStateGuard.Ensure(processPath is not null || !throwOnNull);
        return processPath;
```